### PR TITLE
Set server_announce to world.mt and respect modes on changing subgame

### PR DESCRIPTION
**Pull request by kilbith**
Title says it all.

### CODE REVIEW

File: builtin/mainmenu/tab_singleplayer.lua [hunk 1]
```diff
@@ -38,6 +38,14 @@ local function singleplayer_refresh_gamebar()
...
 					core.setting_set("menu_last_game",gamemgr.games[j].id)
 					menudata.worldlist:set_filtercriteria(gamemgr.games[j].id)
+					local index = filterlist.get_current_index(menudata.worldlist,
+						tonumber(core.setting_get("mainmenu_last_selected_world")))
+					local selected = core.get_textlist_index("sp_worlds")
+					if not index or index < 1 then
+						index = math.min(core.get_textlist_index("sp_worlds"),
+							#menudata.worldlist:get_list())
+					end
+					menu_worldmt_legacy(index)
```
Comment (est31) [inline builtin/mainmenu/tab_singleplayer.lua]: perhaps move this up one line? because otherwise we crash when there is no world, no?

Comment (kilbith) [inline builtin/mainmenu/tab_singleplayer.lua:48]: 1. It doesn't crash if there's no world (tested).
2. Code does not work if I move that one line up (tested).

Comment (est31) [inline builtin/mainmenu/tab_singleplayer.lua]: ah dammit yea you are right
if its "moved up", it would need to get into the else case.

**Comment by est31** [comment]
0a0378fecef5c6b4be6d034d8bced2a1568dde81

**Pull request closed without merging**